src/LibDDRV.sol

```diff
@@ -151,51 +150,84 @@ library LibDDRV {
             // Get weight and range number
             uint256 weight = range.weight;
             uint256 j = floor_ilog(weight) + 1;
-            forest.levels[level].weight += weight;
-            forest.levels[level].roots += j;
             // TODO(Support expanded degree bound)
-            //if (range.children.length > 2) {
-            // Then this range moves to the next level
-            //Node storage newRange = forest.levels[level].ranges[j];
-            //enqueue_range(ptr1, head1, tail1, j, newRange);
-            // Insert range from lower level into this level
-            // insert_bucket()
-            // Delete range from lower level
-            //delete_range(range);
-            //} else {
-            //    forest.levels[level].weight += weight;
-            //    forest.levels[level].roots += j;
-            //}
+            if (range.children.length > 1) {
+                Node storage destRange = insert_range(forest, range, level, level + 1, j);
+                enqueue_range(ptr1, head1, tail1, j, destRange);
+
+                /* TODO: Delete range(?). An explicit call to deleteRange is made in the pseudo, but
```

Comment:
@0xAlcibiades bit confused here; lmk if this comment makes sense or not.

Reply:
It's about migrating afaict. and that is correct, a range on level 2 could have 5 level 0 children.